src/coreHandler.ts

```diff
@@ -75,9 +75,11 @@ export class CoreHandler {
 		})
 		this.core.onDisconnected(() => {
 			this.logger.info('Core Disconnected!')
+			this.setStatus(P.StatusCode.WARNING_MAJOR, ['Core Disconnected'])
```

Comment:
I'm not sure about it. When the connection is lost, how is `setStatus` going to take any effect?